minecraft/1.7/src/main/java/nova/core/wrapper/mc/forge/v17/render/RenderUtility.java

```diff
@@ -129,15 +134,30 @@ public void preTextureHook(TextureStitchEvent.Pre event) {
 	}
 
 	public void registerIcon(Texture texture, TextureStitchEvent.Pre event) {
-		iconMap.put(texture, event.map.registerIcon(texture.getResource()));
+		iconMap.put(texture, event.map.registerIcon(AssetConverter.instance().toNativeTexture(texture).toString()));
 	}
 
 	@SubscribeEvent
 	public void textureHook(TextureStitchEvent.Post event) {
+		Game.render().blockTextures.forEach(this::updateTexureDimensions);
+		Game.render().itemTextures.forEach(this::updateTexureDimensions);
+		Game.render().entityTextures.forEach(this::updateTexureDimensions);
+	}
 
+	private void updateTexureDimensions(Texture texture) {
+		try {
+			Field dimension = Texture.class.getDeclaredField("dimension");
```

Comment:
The `dimension` field is private and I got rid of `RenderManager.getDimension`, because it would cause a crash when the texture was not found.
Also do you have a better idea on how to change the size in the `dimension` field?